x-pack/plugins/infra/server/lib/alerting/inventory_metric_threshold/inventory_metric_threshold_executor.ts

```diff
@@ -39,34 +42,34 @@ type InventoryMetricThresholdAllowedActionGroups = ActionGroupIdsOf<
   typeof FIRED_ACTIONS | typeof WARNING_ACTIONS
 >;
 
-export type InventoryMetricThresholdAlertTypeState = AlertTypeState; // no specific state used
-export type InventoryMetricThresholdAlertInstanceState = AlertInstanceState; // no specific state used
-export type InventoryMetricThresholdAlertInstanceContext = AlertInstanceContext; // no specific instance context used
+export type InventoryMetricThresholdRuleState = RuleTypeState; // no specific state used
+export type InventoryMetricThresholdAlertState = AlertState; // no specific state used
+export type InventoryMetricThresholdAlertContext = AlertContext; // no specific instance context used
 
-type InventoryMetricThresholdAlertInstance = AlertInstance<
-  InventoryMetricThresholdAlertInstanceState,
-  InventoryMetricThresholdAlertInstanceContext,
+type InventoryMetricThresholdAlert = Alert<
+  InventoryMetricThresholdAlertState,
+  InventoryMetricThresholdAlertContext,
   InventoryMetricThresholdAllowedActionGroups
 >;
-type InventoryMetricThresholdAlertInstanceFactory = (
+type InventoryMetricThresholdAlertFactory = (
   id: string,
   reason: string,
   threshold?: number | undefined,
   value?: number | undefined
-) => InventoryMetricThresholdAlertInstance;
+) => InventoryMetricThresholdAlert;
 
 export const createInventoryMetricThresholdExecutor = (libs: InfraBackendLibs) =>
   libs.metricsRules.createLifecycleRuleExecutor<
     InventoryMetricThresholdParams & Record<string, unknown>,
-    InventoryMetricThresholdAlertTypeState,
-    InventoryMetricThresholdAlertInstanceState,
-    InventoryMetricThresholdAlertInstanceContext,
+    InventoryMetricThresholdRuleState,
+    InventoryMetricThresholdAlertState,
+    InventoryMetricThresholdAlertContext,
     InventoryMetricThresholdAllowedActionGroups
   >(async ({ services, params }) => {
     const { criteria, filterQuery, sourceId, nodeType, alertOnNoData } = params;
     if (criteria.length === 0) throw new Error('Cannot execute an alert with 0 conditions');
     const { alertWithLifecycle, savedObjectsClient } = services;
-    const alertInstanceFactory: InventoryMetricThresholdAlertInstanceFactory = (id, reason) =>
```

Comment:
The task_runner of the alerting framework exposes an alertInstanceFactory method https://github.com/elastic/kibana/blob/main/x-pack/plugins/alerting/server/task_runner/task_runner.ts#L321

This `alertInstanceFactory` variable is a local variable in this file, so I don't have we have a problem here. I make the comment just to double check if everything works as expected (I wait for `yarn start` to complete to double check)